### PR TITLE
[fastboot] Notify SAI that fastboot is done

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
**Why I did this**

Notify SAI that fastboot is done

**How I did this**

Set SAI_SWITCH_ATTR_FAST_API_ENABLE to false when fastboot is done

**How I verify this**

Manual test

### CODE REVIEW

**Comment by Junchao-Mellanox** [comment]
Hi @yxieca , @vaibhavhd , could you please kindly review it?

**Review by kcudnik — APPROVED** [review]

**Comment by kcudnik** [comment]
Please fix errors

**Review by stepanblyschak — APPROVED** [review]

**Comment by yxieca** [comment]
@adyeung can you get someone from Broadcom to assess this change?

**Comment by adyeung** [comment]
@tushar-ty please help review

**Comment by Junchao-Mellanox** [comment]
Hi, please hod on the review for a moment. I will need to update this PR.

**Comment by Junchao-Mellanox** [comment]
Hi @tushar-ty @vishnushetty , could you please kindly provide comment?

**Comment by Junchao-Mellanox** [comment]
Hi @adyeung, @tushar-ty, @vishnushetty, kindly reminder.

**Comment by adyeung** [comment]
Adding @Prasanth-KV to help review

**Review by Prasanth-KV — CHANGES_REQUESTED** [review]

File: syncd/Syncd.cpp [hunk 1]
```diff
@@ -3822,7 +3822,8 @@ sai_status_t Syncd::processNotifySyncd(
             m_asicInitViewMode = false;
 
             if (m_commandLineOptions->m_startType == SAI_START_TYPE_FASTFAST_BOOT ||
-                m_commandLineOptions->m_startType == SAI_START_TYPE_EXPRESS_BOOT)
+                m_commandLineOptions->m_startType == SAI_START_TYPE_EXPRESS_BOOT ||
+                m_commandLineOptions->m_startType == SAI_START_TYPE_FAST_BOOT)
```
[the remark this answers is not in the document]
Comment (Junchao-Mellanox) [reply syncd/Syncd.cpp:3826]: Hi @Prasanth-KV , thanks for the comment. We are here to understand if there is any vendor who does not support it. If you know any vendor who does not support it, could you please provide the vendor name? Also, I assume you are from Broadcom, does Broadcom support SAI_SWITCH_ATTR_FAST_API_ENABLE in this case?

[the remark this answers is not in the document]
Comment (Prasanth-KV) [reply syncd/Syncd.cpp]: Broadcom does not support SAI_SWITCH_ATTR_FAST_API_ENABLE. Also, Broadcom doesn't use start type command line option as of now and hence this won't affect Broadcom platforms currently. If SAI_SWITCH_ATTR_FAST_API_ENABLE is required for fast-boot case, it is better to add capability check and skip if not supported.

[the remark this answers is not in the document]
Comment (Junchao-Mellanox) [reply syncd/Syncd.cpp]: thanks! I have pushed the fix, the change will only affect mellanox platform.

**Review by Prasanth-KV — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@Junchao-Mellanox  can you check please the PR checker coverage issue?

**Comment by Junchao-Mellanox** [comment]
> @Junchao-Mellanox can you check please the PR checker coverage issue?

Sure, working on it

**Comment by Junchao-Mellanox** [comment]
The new test case passed locally, but failed in github checker, I will investigate.

**Comment by Junchao-Mellanox** [comment]
Hi @kcudnik , could you please review and merge?

**Review by kcudnik — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@saiarcot895 could you please help to merge?

**Comment by dprital** [comment]
Should be merged only after upstream of SAI version 2405.29.0.0.
@Junchao-Mellanox , please mention it on PR description

**Comment by liat-grozovik** [comment]
@kcudnik can you please help to merge?

**Comment by liat-grozovik** [comment]
@saiarcot895 @kcudnik kindly reminder to merge this one

**Comment by Junchao-Mellanox** [comment]
Hi @kcudnik , could you please help merge this one?

**Comment by bingwang-ms** [comment]
@Junchao-Mellanox Auto-cherry-pick seems not working on this PR. Can you please raise a PR to 202405 branch?

**Comment by Junchao-Mellanox** [comment]
Backport PR: https://github.com/sonic-net/sonic-sairedis/pull/1468

**Comment by zhangyanzhao** [comment]
proposed for 202411 release but cherry pick to 202405.

**Pull request merged**